CHANGELOG.rst

```diff
@@ -1,3 +1,27 @@
+
+6.3.0
+=======
+
+.. warning::
+
+   This release explicitly warns on unsupported setuptools.
+   This unfortunately has to happen as the legacy ``setup_requires`` mechanism
+   incorrectly configures the setuptools working-set when a more recent setuptools
+   version than available is required.
+
+   As all releases of setuptools are affected as the historic mechanism
+   for ensuring a working setuptools setup was shipping a ``ez_setup`` file
+   next to ``setup.py``, which would install the required version of setuptools.
+
+   This mechanism has long since been deprecated and removed
+   as most people haven't been using it
```

Comment:
```suggestion
   as most people haven't been using it.
```

Reply:
Also, on this block, mentioning what version is too old and going to produce the warning might be helpful. This starts by saying "warns on legacy" but then says "all releases are affected", it's really unclear, I think. Maybe I could suggest a new version?


```rst
This release explicitly warns on unsupported setuptools. When using the legacy 
``setup_requires`` mechanism, you cannot request a newer setuptools than the
one currently running (this was the reason PEP 517 and PEP 518 were written).
```

Reply:
all setuptools releases are affected by the issue, even those that dont trigger legacy problems